labels/index.ts

```diff
@@ -0,0 +1,109 @@
+import './style.css';
+import 'maplibre-gl/dist/maplibre-gl.css';
+import {Deck} from '@deck.gl/core';
+import maplibregl from 'maplibre-gl';
+import {VectorTileLayer, vectorQuerySource} from '@deck.gl/carto';
+import {CollisionFilterExtension} from '@deck.gl/extensions';
+import {scaleLinear} from 'd3-scale';
+
+const apiBaseUrl = import.meta.env.VITE_API_BASE_URL;
+const accessToken = import.meta.env.VITE_API_ACCESS_TOKEN;
+const connectionName = 'carto_dw';
+const cartoConfig = {apiBaseUrl, accessToken, connectionName};
+
+const INITIAL_VIEW_STATE = {
+  latitude: 39.1,
+  longitude: -94.57,
+  zoom: 3.8,
+  maxZoom: 16,
+  pitch: 0,
+  bearing: 0
+};
+
+const colorScale = scaleLinear()
+  .domain([3, 4, 5, 6, 7])
+  .range([
+    [29, 145, 192],
+    [65, 182, 196],
+    [127, 205, 187],
+    [199, 233, 180],
+    [237, 248, 177]
+  ]);
+
+  const fontSize = 24;
+  const scale = 16;
+  const sizeMaxPixels = (scale / 3) * fontSize;
+  const sizeMinPixels = Math.min(scale / 1000, 0.5) * fontSize;
+
+const MAP_STYLE = 'https://basemaps.cartocdn.com/gl/dark-matter-nolabels-gl-style/style.json';
+
+const deck = new Deck({
+  canvas: 'deck-canvas',
+  initialViewState: INITIAL_VIEW_STATE,
+  controller: true
+});
+
+// Add basemap
+const map = new maplibregl.Map({
+  container: 'map',
+  style: MAP_STYLE,
+  interactive: false
+});
+
+deck.setProps({
+  onViewStateChange: ({viewState}) => {
+    const {longitude, latitude, ...rest} = viewState;
+    map.jumpTo({center: [longitude, latitude], ...rest});
+  }
+});
+
+let selectedOsmCategory = 'Financial';
+const osmCategorySelector = document.querySelector<HTMLSelectElement>('#osmCategorySelector');
+osmCategorySelector?.addEventListener('change', () => {
+  selectedOsmCategory = osmCategorySelector.value;
+  render();
+});
+
+async function render() {
+  const dataSource = vectorQuerySource({
+    ...cartoConfig,
+    sqlQuery: `select geom, population, name from cartobq.public_account.cities_1000 WHERE population > 0 ORDER BY population DESC`
```

Comment:
Format this better:

SELECT geom... FROM

Could you explain why is ORDER BY required?

Reply:
Formatted, and ORDER is to show the bigest population cities to apply in collision priority